types/wlr_relative_pointer_v1.c

```diff
@@ -0,0 +1,192 @@
+		return;
+	}
+
+	relative_pointer->client = client;
+	relative_pointer->resource = relative_pointer_resource;
+	relative_pointer->pointer = (struct wl_pointer *) pointer;
+
+	wl_signal_init(&relative_pointer->events.destroy);
+
+	wl_list_insert(&seat_client->relative_pointers,
+		wl_resource_get_link(relative_pointer_resource));
+
+	wl_resource_set_implementation(relative_pointer_resource, &relative_pointer_v1_impl,
+		relative_pointer, relative_pointer_v1_handle_resource_destroy);
+
+	struct wlr_relative_pointer_manager_v1 *relative_pointer_manager =
+		wl_resource_get_user_data(resource);
+	wlr_signal_emit_safe(&relative_pointer_manager->events.new_relative_pointer,
+		relative_pointer);
+
+	wlr_log(WLR_DEBUG, "relative_pointer_v1 %p created for client %p",
+		relative_pointer, client);
+}
+
+
+static void relative_pointer_v1_handle_destroy(struct wl_client *client,
+		struct wl_resource *resource) {
+	struct wlr_relative_pointer_v1 *relative_pointer =
+		wl_resource_get_user_data(resource);
+	wlr_log(WLR_DEBUG, "relative_pointer_v1 %p released by client %p",
+		relative_pointer, client);
+
+	wl_resource_destroy(resource);
+}
+
+
+static void relative_pointer_manager_v1_bind(struct wl_client *wl_client, void *data,
+		uint32_t version, uint32_t id) {
+	struct wlr_relative_pointer_manager_v1 *relative_pointer_manager = data;
+
+	struct wl_resource *manager_resource = wl_resource_create(wl_client,
+		&zwp_relative_pointer_manager_v1_interface, version, id);
+
+	if (manager_resource == NULL) {
+		wl_client_post_no_memory(wl_client);
+		return;
+	}
+
+	wl_list_insert(&relative_pointer_manager->resources, wl_resource_get_link(manager_resource));
+
+	wl_resource_set_implementation(manager_resource, &relative_pointer_manager_v1_impl,
+		relative_pointer_manager, relative_pointer_manager_v1_handle_resource_destroy);
+
+	wlr_log(WLR_DEBUG, "relative_pointer_v1 manager bound to client %p",
+		wl_client);
+}
+
+
+/**
+ * Implementations
+ */
+
+static const struct zwp_relative_pointer_manager_v1_interface relative_pointer_manager_v1_impl = {
+	.destroy = relative_pointer_manager_v1_handle_destroy,
+	.get_relative_pointer = relative_pointer_manager_v1_handle_get_relative_pointer,
+};
+
+
+static const struct zwp_relative_pointer_v1_interface relative_pointer_v1_impl = {
+	.destroy = relative_pointer_v1_handle_destroy,
+};
+
+
+/**
+ * Public functions
+ */
+
+struct wlr_relative_pointer_manager_v1 *wlr_relative_pointer_v1_create(struct wl_display *display) {
+	struct wlr_relative_pointer_manager_v1 *relative_pointer_manager =
+		calloc(1, sizeof(struct wlr_relative_pointer_manager_v1));
+
+	if (relative_pointer_manager == NULL) {
+		return NULL;
+	}
+
+	wl_list_init(&relative_pointer_manager->resources);
+
+	wl_signal_init(&relative_pointer_manager->events.destroy);
+	wl_signal_init(&relative_pointer_manager->events.new_relative_pointer);
+
+	relative_pointer_manager->global = wl_global_create(display,
+		&zwp_relative_pointer_manager_v1_interface, 1,
```

Comment:
Is it possible to define the version with `RELATIVE_POINTER_MANAGER_VERSION` at the beginning of the file? It makes it way easier to locate it.

Reply:
Should it be undef'ed at the end of the file?

Reply:
Since this is a `.c` file, it doesn't need to, it won't be included in another file anyway.